source/client/output_formatter_impl.cc

```diff
@@ -288,45 +287,58 @@ std::string FortioOutputFormatterImpl::formatProto(const nighthawk::client::Outp
   fortio_output.set_numthreads(number_of_connections);
 
   // Get the result that represents all workers (global)
-  const auto& nh_global_result = this->getGlobalResult(output);
+  const auto& nh_global_result = getGlobalResult(output);
 
   // Fill in the actual QPS based on the counters
-  const auto& nh_rq_counter = this->getCounterByName(nh_global_result, "upstream_rq_total");
-  const double actual_qps = static_cast<double>(
-      nh_rq_counter.value() / std::chrono::duration<double>(actual_duration).count());
-  fortio_output.set_actualqps(actual_qps);
+  if (!visitCounter(
+          nh_global_result, "upstream_rq_total",
+          [&fortio_output, actual_duration](const nighthawk::client::Counter& nh_counter) {
+            const double actual_qps = static_cast<double>(
+                nh_counter.value() / std::chrono::duration<double>(actual_duration).count());
+            fortio_output.set_actualqps(actual_qps);
+          })) {
+    throw NighthawkException(
+        "Nighthawk result was malformed, contains no counter with name: upstream_rq_total");
+  };
+  visitCounter(nh_global_result, "upstream_cx_rx_bytes_total",
+               [&fortio_output](const nighthawk::client::Counter& nh_counter) {
+                 fortio_output.set_bytesreceived(nh_counter.value());
+               });
+  visitCounter(nh_global_result, "upstream_cx_tx_bytes_total",
+               [&fortio_output](const nighthawk::client::Counter& nh_counter) {
+                 fortio_output.set_bytessent(nh_counter.value());
+               });
 
   // Fill in the number of successful responses.
   // Fortio-ui only reads the 200 OK field, other fields are never displayed.
+  // If this field doesn't exist, then there were no 2xx responses
   fortio_output.mutable_retcodes()->insert({"200", 0});
-  try {
-    const auto& nh_2xx_counter = this->getCounterByName(nh_global_result, "benchmark.http_2xx");
-    // So Fortio computes the error percentage based on:
-    // - the sample count in the histogram
-    // - the number of 200 responses
-    // Nighthawk workers perform a single-request as a warmup, and doesn't measure latency for that.
-    // So we do an approximation here: we substract number_of_workers from the observed 2xx
-    // responses.
-    // TODO(oschaaf): It would be better to compute the actual ratio of error codes vs success
-    // codes.. and possibly also factor in connection failures, etc.
-    fortio_output.mutable_retcodes()->at("200") = nh_2xx_counter.value() - number_of_workers;
-  } catch (const NighthawkException& e) {
-    // If this field doesn't exist, then there were no 2xx responses
-    fortio_output.mutable_retcodes()->at("200") = 0;
-  }
-
-  auto* statistic =
-      this->findStatistic(nh_global_result, "benchmark_http_client.request_to_response");
+  visitCounter(nh_global_result, "benchmark.http_2xx",
```

Comment:
The behavior of the new code seems different -- if the counter isn't found, the old code used to insert a 0 under the "200" retcode. Is it safe to assume the `benchmark.http_2xx` counter will always be present?

Reply:
> The behavior of the new code seems different -- if the counter isn't found, the old code used to insert a 0 under the "200" retcode. Is it safe to assume the `benchmark.http_2xx` counter will always be present?

Well, I came across this, and noticed that it gets inserted in the line above unconditionally. So I figured it wasn't needed below, and intended to clean that up. Does that make sense?

Reply:
Ohh I see, yes, this one should be ok.